cw-orch/src/interchain/interchain_channel_builder.rs

```diff
@@ -0,0 +1,233 @@
+use crate::interchain::docker::DockerHelper;
+use crate::interchain::interchain_channel::InterchainChannel;
+use crate::queriers::Ibc;
+use crate::{DaemonError, Hermes};
+
+use super::interchain_channel::InterchainPort;
+
+
+#[derive(Default)]
+struct ChainChannelBuilder{
+	pub chain_id: Option<String>,
+	pub port: Option<String>,
+	pub grpc: Option<Grpc>,
+	pub is_local_chain: Option<bool>
+}
+
+
+impl ChainChannelBuilder{
+	async fn create_grpc_channel(&self) -> Result<Channel, DaemonError>{
+		let grpcs = if let Some(grpc) = &self.grpc{
+			vec![grpc.clone()]
+		}else{
+			// If the GRPC is not registered, we try querying the chain from the local configuration ?
+			// TODO
+		    let mut chains: Vec<ChainData> = vec![parse_network(&self.chain_id.clone().unwrap()).into()];
+		    if self.is_local_chain.unwrap_or(false){
+			    // This is only for tests, we may change that later ?
+			    InterchainInfrastructure::configure_networks(&mut chains).await?;
+			}
+
+		    chains[0].apis.grpc.clone()
+		};
+		GrpcChannel::connect(&grpcs, &ChainId::from_string(&self.chain_id.clone().unwrap())).await
+	}
+}
+
+
+#[derive(Default)]
+pub struct InterchainChannelBuilder{
+	chain_a: ChainChannelBuilder,
+	chain_b: ChainChannelBuilder,
+	connection_a: Option<String>,
+}
+
+impl InterchainChannelBuilder{
+
+    async fn get_hermes() -> IcResult<Hermes> {
+        let docker_helper = DockerHelper::new().await?;
+        docker_helper.get_hermes()
+    }
+
+	pub fn chain_a(&mut self, chain_id: impl Into<String>) -> &mut Self{
+		self.chain_a.chain_id = Some(chain_id.into());
+		self
+	}
+
+	pub fn chain_b(&mut self, chain_id: impl Into<String>) -> &mut Self{
+		self.chain_b.chain_id = Some(chain_id.into());
+		self
+	}
+
+	pub fn add_grpc_a(&mut self, grpc: Grpc) -> &mut Self{
+		self.chain_a.grpc = Some(grpc);
+		self
+	}
+
+	pub fn add_grpc_b(&mut self, grpc: Grpc) -> &mut Self{
+		self.chain_b.grpc = Some(grpc);
+		self
+	}
+
+	pub fn is_local_chain_a(&mut self) -> &mut Self{
+		self.chain_a.is_local_chain = Some(true);
+		self
+	}
+
+	pub fn is_local_chain_b(&mut self) -> &mut Self{
+		self.chain_b.is_local_chain = Some(true);
+		self
+	}
+
+	pub fn port_a(&mut self, port: impl Into<String>) -> &mut Self{
+		self.chain_a.port = Some(port.into());
+		self
+	}
+
+	pub fn port_b(&mut self, port: impl Into<String>) -> &mut Self{
+		self.chain_b.port = Some(port.into());
+		self
+	}
+
+	pub fn connection(&mut self, connection: impl Into<String>) -> &mut Self{
+		self.connection_a = Some(connection.into());
+		self
+	}
+
+	// The channel id id supposed to be the one created on the a side (you can interchange a and b at will to allow for that)
+	// The connection between the 2 chains, The two chains_ids, and the port of chain a should be defined when building a channel object with this method
+	// TODO enforce that with errors ? 
+	pub async fn channel_from(&self, channel_id_a: String) -> Result<InterchainChannel, DaemonError>{
```

Comment:
This functions creates an `InterchainChannel` object from an existing channel between two ports

Reply:
In order to work, this function needs /
1. the full description of the first port (chain_id, grpc channel, port, channel) 
2. a partial description of the second port (chain id, grpc_channel)